test/decl/ext/typealias.swift

```diff
@@ -78,3 +78,13 @@ extension FooIntBarFloatDoubleInner {
   }
 }
 
+struct Foo2<T1, T2, T3,> {}
+
+typealias Bar2<
+  T1,
+  T2,
+> = Foo2<
+  T1,
+  T2,
+  Bool,
+>
```

Comment:
Would you mind adding test cases for generic arguments in expression positions to prevent future regressions? Like `let _ = Generic<Int, Bool,>.self` and `let _ = Generic<Int, Bool,>()`. It also uses `parseGenericArguments()` so it should work with your changes.

Reply:
good idea, added some examples like that

Reply:
Thanks for remembering to check that! It didn't work originally, we also had to update the lookahead code that disambiguated the `<` between a generic list and a less than operator. Updated, working now :)

Reply:
Ah `canParseGenericArguments()`. That makes sense. Thank you for fixing it!